src/libstore/outputs-spec.cc

```diff
@@ -21,7 +21,8 @@ bool OutputsSpec::contains(const std::string & outputName) const
 
 std::optional<OutputsSpec> OutputsSpec::parseOpt(std::string_view s)
 {
-    static std::regex regex(R"((\*)|([a-z]+(,[a-z]+)*))");
+    // See checkName() for valid output name characters.
+    static std::regex regex(R"((\*)|([a-zA-Z\+\-\._\?=]+(,[a-zA-Z\+\-\._\?=]+)*))");
```

Comment:
Can be in a future PR, but it would be great for these two to share code so they are automatically in sync. Maybe we could `constexpr` build the regex from one to just match a single output name used in `checkName`, for example.

Reply:
I've checked the regex and it looks accurate.